sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -524,6 +524,22 @@ def add_parent(child, parent):
         add_parent(parent, parents.get(parent))
       components.transforms[parent].subtransforms.append(child)
 
+  copied_subtransform_ids = set()
```

Comment:
Given that the set of subtransformation is a tree (no cycles) we can get rid of this.

Reply:
Thanks! This simplifies things.